### ISSUE TITLE
is it fully invertible? last layer is pooling+ linear

### ISSUE THREAD

**Issue by tsauri** [jhjacobsen/pytorch-i-revnet#9]
Seems that the network is only 90% invertible. last layer relies on pooling and linear layer.
Can we replace pooling+relu+linear layer with more rev convnet downsampling and 1x1 reversible convolution?
@jhjacobsen  did you tested this?

**Comment by jhjacobsen** [comment]
We have follow-up work where we also make the very last layer invertible by reading out the logits in the final invertible representation: https://arxiv.org/abs/1811.00401

**Comment by sobalgi** [comment]
Hello @jhjacobsen,

Your follow-up work on understanding the caveats in the existing deep learning architectures is really fascinating. I could not understand what you meant by "read out the logits from low pass components" as DCT on the features of final layer yeilds the output of same size. I'm not sure what you meant by readout logits.

Can you please point towards the implementation of the fiRevNets or Indicate how exactly we can replace the last layer to obtain logits instead of feature maps ?

Many thanks,
@sobalgi.
